articles/Object-Detection-using-Fast-R-CNN-BrainScript.md

```diff
@@ -0,0 +1,537 @@
+</p>
+
+While this approach works well in terms of accuracy, it is very costly to compute since the Neural Network has to be evaluated
+for each ROI. Fast R-CNN addresses this drawback by only evaluating most of the network (to be specific: the convolution layers)
+a single time per image. According to the authors, this leads to a 213 times speed-up during testing and a 9x speed-up during
+training without loss of accuracy. This is achieved by using an ROI pooling layer which projects the ROI onto the convolutional
+feature map and performs max pooling to generate the desired output size that the following layer is expecting.
+In the AlexNet example used in this tutorial the ROI pooling layer is put between the last convolutional layer and the first
+fully connected layer (see [BrainScript code](#cntk-configuration)).
+
+The original Caffe implementation used in the R-CNN papers can be found at GitHub:
+[RCNN](https://github.com/rbgirshick/rcnn), [Fast R-CNN](https://github.com/rbgirshick/fast-rcnn), and [Faster R-CNN](https://github.com/rbgirshick/py-faster-rcnn). This tutorial uses some of the code from these repositories, notably (but not exclusively) for SVM training and model evaluation.
+
+#### SVM vs NN training
+
+Patrick Buehler provides instructions on how to train an SVM on the CNTK Fast R-CNN output (using the 4096 features from the last fully connected layer)
+as well as a discussion on pros and cons [here](https://github.com/Azure/ObjectDetectionUsingCntk).
+
+### Selective Search
+[Selective Search](http://koen.me/research/pub/uijlings-ijcv2013-draft.pdf) is a method for finding a large set of possible
+object locations in an image, independent of the class of the actual object. It works by clustering image pixels into segments,
+and then performing hierarchical clustering to combine segments from the same object into object proposals.
+
+<p align="center">
+<a target="_blank" href="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.noGridNoFiltering.roi.jpg"><img src="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.noGridNoFiltering.roi.jpg" alt="image" height="300"/></a>
+<a target="_blank" href="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.noGrid.roi.jpg"><img src="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.noGrid.roi.jpg" alt="image" height="300"/></a>
+<a target="_blank" href="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.roi.jpg"><img src="./Tutorial_FastRCNN/WIN_20160803_11_29_07_Pro.roi.jpg" alt="image" height="300"/></a>
+</p>
+
+To complement the detected ROIs from Selective Search, we add ROIs that uniform cover the image at different scales and aspect ratios.
+The image on the left shows an example output of Selective Search, where each possible object location is visualized by a green
+rectangle. ROIs that are too small, too big, etc. are discarded (middle) and finally ROIs that uniformly cover the image are added (right).
```

Comment:
>right [](start = 132, length = 5)

it seems like middle left right etc are depending on your screen resolution?

Reply:
replaced by first, second and third to account for resolution